Getters/setters shouldn't trigger no-dupe-keys

The following code triggers the no-dupe-keys rule in v0.4.0:

``` javascript
var foo = {
    get prop() {
        return this.p;
    },
    set prop(val) {
        this.p = val;
    }
};
```

``` bash
$ eslint foo.js

foo.js
  1:10  error  Duplicate key 'prop'            no-dupe-keys
  2:12  error  Missing "use strict" statement  strict
  5:15  error  Missing "use strict" statement  strict

✖ 3 problems
```

It seems that the rule doesn't consider getters/setters declared in object initializers.

--- a/lib/rules/no-dupe-keys.js
+++ b/lib/rules/no-dupe-keys.js
@@ -20,14 +20,13 @@ module.exports = function(context) {
             var nodeProps = {};
 
             node.properties.forEach(function(property) {
+                var keyName = property.key.name || property.key.value;
+                var key = property.kind + "-" + keyName;
 
-                // Create a safe key by prefixing with "$-".
-                var safeKey = "$-" + (property.key.name || property.key.value);
-
-                if (nodeProps[safeKey]) {
-                    context.report(node, "Duplicate key '{{key}}'.", { key: safeKey.split("-")[1] });
+                if (nodeProps[key]) {
+                    context.report(node, "Duplicate key '{{key}}'.", { key: keyName });
                 } else {
-                    nodeProps[safeKey] = true;
+                    nodeProps[key] = true;
                 }
             });
 

--- a/tests/lib/rules/no-dupe-keys.js
+++ b/tests/lib/rules/no-dupe-keys.js
@@ -16,7 +16,8 @@ var eslintTester = require("eslint-tester");
 eslintTester.addRuleTest("lib/rules/no-dupe-keys", {
     valid: [
         "var foo = { __proto__: 1, two: 2};",
-        "var x = { foo: 1, bar: 2 };"
+        "var x = { foo: 1, bar: 2 };",
+        "+{ get a() { }, set a(b) { } };"
     ],
     invalid: [
         { code: "var x = { y: 1, y: 2 };", errors: [{ message: "Duplicate key 'y'.", type: "ObjectExpression"}] },

